src/TMS/Wizard/Player.php

```diff
@@ -299,7 +300,15 @@ protected function finish(State $state) {
 		for ($i = 0; $i < count($steps); $i++) {
 			$step = $steps[$i];
 			$data = $state->getStepData($i);
-			$message = $step->processStep($data);
+			try {
+				$message = $step->processStep($data);
+			} catch (OverbookedException $e) {
+				$this->state_db->delete($state);
+				$this->wizard->finish();
+				$this->ilias_bindings->redirectToPreviousLocation(array($this->ilias_bindings->txt($e->getMessage())), false);
+				return;
+			}
```

Comment:
Das ist hier falsch. In `TMS\Wizard` befindet sich die allgemeine Implementierung für einen Wizard. So bekommt sie eine Abhängigkeit gegen `TMS\Booking`. Der Pfeil geht aber eigentlich anders rum: `Booking` hängt von `Wizard` ab.